src/NLog/LayoutRenderers/AllEventPropertiesLayoutRenderer.cs

```diff
@@ -223,7 +223,7 @@ private bool CheckForExclude(LogEventInfo logEvent)
                 checkForExclude = !(logEvent.CallSiteInformation == null && Exclude.Count == CallerInformationAttributeNames.Length && Exclude.Contains(CallerInformationAttributeNames[0]));
             }
 #endif
-            return checkForExclude && logEvent.HasProperties;
```

Comment:
double check, this check isn't needed? (probably check in the calling method?)

Reply:
No it has already been checked before calling this method.